### PR TITLE
add suport for REBAR_LIB_DEPS defined but not used situation

### PR DESCRIPTION
**Pull request by no-trace-of-time**
use case:
1.REBAR_LIB_DEPS env var defined
2. parent project do not want to use external lib dir
current result:
1. internal_otp_vsn.hrl will generated in $REBAR_LIB_DEPS/otp_vsn/include/internal_otp_vsn.hrl, not in _build/lib/otp_vsn/include/internal_otp_vsn.hrl
2. result compile error , not found internal_otp_vsn.hrl for parent project

solution:
1. write_file on both possible dirs: _build/lib/otp_vsn/include and $REBAR_LIB_DEPS/otp_vsn/inlcude (in env defined)

### CODE REVIEW

**Comment by no-trace-of-time** [comment]
extra:
1.add erlang 21.1 version support

File: .gitignore [hunk 1]
```diff
@@ -16,7 +16,7 @@ _build
 .idea
 *.iml
 rebar3.crashdump
-
+rebar.lock
```
Comment (fenollp) [inline .gitignore]: Please revert this & keep the lockfile.

File: rebar.config.script [hunk 2]
```diff
@@ -46,24 +46,33 @@ begin
       end,
```
Comment (fenollp) [inline rebar.config.script:46]: Also leave files on chmod 644. How did they get to chmod 755?

File: rebar.config.script [hunk 3]
```diff
@@ -46,24 +46,33 @@ begin
...
+                    [LibDir,"otp_vsn","include","internal_otp_vsn.hrl"])
+                    || LibDir <- DepsDirs
+                ];
             true ->
                 %% Current app is this app itself
-                filename:join("include", "internal_otp_vsn.hrl")
+                [filename:join("include", "internal_otp_vsn.hrl")]
         end,
-  ok = file:write_file(Dst, Data),
+  %ok = file:write_file(Dst, Data),
```
Comment (fenollp) [inline rebar.config.script:73]: Please remove this line.

**Review by fenollp — CHANGES_REQUESTED** [review]
Appart from my few comments: LGTM! Thanks :)

File: rebar.config.script (hunk 3 above)
Comment (no-trace-of-time) [reply rebar.config.script:73]: has already changed to list comprehension

**Comment by no-trace-of-time** [comment]
Requested changes have been made.

**Comment by fenollp** [comment]
Thanks. Will publish a minor release later today

**Pull request merged**